bpf/estat/backend-io.c

```diff
@@ -40,7 +40,7 @@ disk_io_start(struct pt_regs *ctx, struct request *reqp)
 	data.ts = bpf_ktime_get_ns();
 	data.cmd_flags = reqp->cmd_flags;
 	data.size = reqp->__data_len;
-	__builtin_memcpy(&data.device, diskp->disk_name, DEV_NAME_LEN);
+	bpf_probe_read_str(&data.device, DEV_NAME_LEN, diskp->disk_name);
```

Comment:
We should also try changing this to `DISK_NAME_LEN` (and corresponding field `device` of `io_data_t`) to match the `io.st`

Reply:
`DEV_NAME_LEN` (defined here) is 8, while `DISK_NAME_LEN` (defined in the kernel) is 32. I tried using `DISK_NAME_LEN` instead, and `backend-io` seems to run fine with various options enabled. I'm not sure why we chose to use a smaller value here, though.

Reply:
Great, thanks for checking on this, we should go with the kernel defined one (`DISK_NAME_LEN`) then.

IIRC, during our early development there were issues with the string/buffer sizes when testing with various options and hence had ended up with the smaller size.